File: widget/src/main/java/id/mochadwi/cataloguewidget/model/BaseResponse.java
package io.mochadwi.cataloguewidget.model;

import android.support.annotation.Nullable;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class BaseResponse<T> {

    @SerializedName("dates")
    private Dates dates;

    @SerializedName("page")
    private int page;

    @SerializedName("total_pages")
    private int totalPages;

    @SerializedName("results")
    private List<T> results;

    @SerializedName("total_results")
    private int totalResults;

    public Dates getDates() {
        return dates;
    }

    public void setDates(Dates dates) {
        this.dates = dates;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(int totalPages) {
        this.totalPages = totalPages;
    }

    @Nullable
    public List<T> getResults() {
        return results;
    }

    public void setResults(List<T> results) {
        this.results = results;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(int totalResults) {
        this.totalResults = totalResults;
    }

    @Override
    public String toString() {
        return new Gson().toJson(this);
    }

    public class Dates {

        @SerializedName("maximum")
        private String maximum;

        @SerializedName("minimum")
        private String minimum;

        public String getMaximum() {
            return maximum;
        }

        public void setMaximum(String maximum) {
            this.maximum = maximum;
        }

        public String getMinimum() {
            return minimum;
        }

        public void setMinimum(String minimum) {
            this.minimum = minimum;
        }

        @Override
        public String toString() {
            return new Gson().toJson(this);
        }
    }
}

File: favourite/src/main/java/io/mochadwi/cataloguefavourite/model/BaseResponse.kt
package io.mochadwi.cataloguefavourite.model

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

class BaseResponse<T> {

    @SerializedName("dates")
    var dates: Dates? = null

    @SerializedName("page")
    var page: Int = 0

    @SerializedName("total_pages")
    var totalPages: Int = 0

    @SerializedName("results")
    var results: List<T>? = null

    @SerializedName("total_results")
    var totalResults: Int = 0

    override fun toString(): String {
        return Gson().toJson(this)
    }

    inner class Dates {

        @SerializedName("maximum")
        var maximum: String? = null

        @SerializedName("minimum")
        var minimum: String? = null

        override fun toString(): String {
            return Gson().toJson(this)
        }
    }
}